### PR TITLE
[integ-tests] improve build-image integration tests

### PR DESCRIPTION
**Pull request by hanwen-cluster**
### Description of changes
* The first 3 commits are cherry-pick from `release-3.12` branch.
* The last 3 commits are to make sure kernel consistency on RHEL/Rocky build-image

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/common/utils.py [hunk 1]
```diff
@@ -52,8 +52,8 @@
     "rocky8": {"name": "Rocky-8-EC2-Base-8.8*", "owners": ["792107900819"]},  # TODO add china and govcloud accounts
     "rhel8.9": {"name": "RHEL-8.9*_HVM-*", "owners": RHEL_OWNERS},
     "rocky8.9": {"name": "Rocky-8-EC2-Base-8.9*", "owners": ["792107900819"]},  # TODO add china and govcloud accounts
-    "rhel9": {"name": "RHEL-9.3*_HVM-*", "owners": RHEL_OWNERS},
-    "rocky9": {"name": "Rocky-9-EC2-Base-9.3*", "owners": ["792107900819"]},  # TODO add china and govcloud accounts
+    "rhel9": {"name": "RHEL-9.*_HVM*", "owners": RHEL_OWNERS},
```
Comment (dreambeyondorange) [inline tests/integration-tests/tests/common/utils.py:55]: should we be pinning to 9.4 until fsx lustre supports 9.5?

File: tests/integration-tests/tests/createami/test_createami/test_build_image/image.config.yaml [hunk 2]
```diff
@@ -14,17 +14,14 @@ Build:
...
-{% endif %}
-{% if os in ["ubuntu2204", "rhel9", "rocky9"] %}
-# Disable Lustre installation because these newer operating systems release new kernels more often. Lustre usually does not support the latest kernels
+        Enabled: {{ update_os_packages }}
     Installation:
         LustreClient:
-            Enabled: false
-{% endif %}
+            # Disable Lustre installation because these newer operating systems release new kernels more often. Lustre usually does not support the latest kernels
+            Enabled: {% if os in ["ubuntu2204", "rhel9", "rocky9"] %} false {% else %} true {% endif %}
```
Comment (dreambeyondorange) [inline tests/integration-tests/tests/createami/test_createami/test_build_image/image.config.yaml]: i wonder if we even care about 9.5 until this is supported.

File: tests/integration-tests/tests/common/utils.py (hunk 1 above)
Comment (hanwen-cluster) [reply tests/integration-tests/tests/common/utils.py:55]: This test has Lustre installation disabled for RHEL/Rocky. So we can go head

File: tests/integration-tests/tests/createami/test_createami/test_build_image/image.config.yaml (hunk 2 above)
Comment (hanwen-cluster) [reply tests/integration-tests/tests/createami/test_createami/test_build_image/image.config.yaml]: I agree. Anyway it is good to make the tests more robust

File: tests/integration-tests/tests/common/utils.py (hunk 1 above)
Comment (dreambeyondorange) [reply tests/integration-tests/tests/common/utils.py:55]: I'm not sure if this is what we want.  We aren't going to release for an image that doesn't support lustre, so whats the point of testing it?  I think we should enable lustre and stick with supported OSes

File: tests/integration-tests/tests/createami/test_createami/test_build_image/image.config.yaml (hunk 2 above)
Comment (dreambeyondorange) [reply tests/integration-tests/tests/createami/test_createami/test_build_image/image.config.yaml]: lustre should just always be enabled i think

Comment (hanwen-cluster) [reply tests/integration-tests/tests/createami/test_createami/test_build_image/image.config.yaml]: Done, added another commit

**Review by dreambeyondorange — APPROVED** [review]

**Pull request merged**